LockExample/ResultInfo.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace LockExample
+{
+    internal sealed class ResultInfo
```

Comment:
it is possible to make internal classes accessible to unit tests: https://stackoverflow.com/questions/358196/c-sharp-internal-access-modifier-when-doing-unit-testing